pkg/api/customization/persistentvolumeclaim/validator.go

```diff
@@ -27,10 +28,14 @@ func (v *Validator) Validator(request *types.APIContext, schema *types.Schema, d
 	}
 
 	storageClass, err := c.Storage.StorageClasses("").Get(storageClassID, v1.GetOptions{})
-	if err != nil {
+	if err != nil && !apierrors.IsNotFound(err) {
 		return err
 	}
 
+	if apierrors.IsNotFound(err) || storageClass == nil {
```

Comment:
We don't need storageClass only if the specified persistentVolume exists right, so before returning nil should we add a check to see if the PV corresponding to the `volumeId` from the request exists or not?

Reply:
No a persistentVolumeClaim can be created without either the storageClass or an existing persistentVolume - both of these are used in order to match the pvc to a pv if it exists. If none of them are specified, the pvc remains in "pending"/"unbound" state.

Reply:
I added an integration test for this case.